geoapps/utils/utils.py

```diff
@@ -41,6 +41,37 @@
 from sklearn.neighbors import KernelDensity
 
 
+def sorted_children_dict(object: UUID | Entity):
+
+    if isinstance(object, UUID):
+        object = workspace.get_entity(object)[0]
+
+    children_dict = {}
+    iters = []
+    iteration_data = []
+    other_data = []
+    for c in object.children:
+        if not isinstance(c, (IntegerData, FloatData)):
+            continue
+        else:
+            name = c.name
+            children_dict[name] = c.uid
+            iter = re.findall(r"[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?", name)
+            if iter:
+                try:
+                    iters.append(int(iter[0]))
+                except:
+                    iters.append(float(iter[0]))
+                iteration_data.append(name)
+            else:
+                other_data.append(name)
+
+    iteration_data = np.array(iteration_data)[np.argsort(iters)].tolist()
```

Comment:
What would happen if the object has two sets of names with numerical value in it? Let's say we have Obs[1], Obs[2], ... and Pred[1], Pred[2] ... Looks like we are going to mix them both together the way it's done now? Feels like we need to keep track of the unique base names...

Reply:
handled